src/main/java/mate/academy/internetshop/dao/hibernate/OrderDaoHibernateImpl.java

```diff
@@ -0,0 +1,112 @@
+import mate.academy.internetshop.model.User;
+import mate.academy.internetshop.util.HibernateUtil;
+import org.apache.log4j.Logger;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class OrderDaoHibernateImpl implements OrderDao {
+    private static final Logger logger = Logger.getLogger(OrderDaoHibernateImpl.class);
+
+    @Inject
+    private static UserDao userDao;
+
+    @Override
+    public Order create(Order order) {
+        Long orderId = null;
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            orderId = (Long) session.save(order);
+            transaction.commit();
+        } catch (Exception e) {
+            logger.error("Can't create Order " + order, e);
+            if (transaction != null) {
+                transaction.rollback();
+            }
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        order.setId(orderId);
+        return order;
+    }
+
+    @Override
+    public Order get(Long id) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Order order = session.get(Order.class, id);
+            return order;
+        }
+    }
+
+    @Override
+    public Order update(Order order) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            transaction = session.beginTransaction();
+            session.update(order);
+            transaction.commit();
+        } catch (Exception e) {
+            logger.error("Can't update Order " + order, e);
+            if (transaction != null) {
+                transaction.rollback();
+            }
+        }
+        return order;
+    }
+
+    @Override
+    public Order delete(Long id) {
+        Order order = get(id);
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.delete(order);
+            transaction.commit();
+        } catch (Exception e) {
+            logger.error("Can't delete order with id " + id, e);
+            if (transaction != null) {
+                transaction.rollback();
+            }
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return order;
+    }
+
+    @Override
+    public List<Order> getOrders(Long userId) {
+        List<Order> list = new ArrayList<>();
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query query = session.createQuery("FROM Order WHERE user=:user");
+            User user = userDao.get(userId);
```

Comment:
@boroda4436
`
Query query = session.createQuery("FROM Order WHERE userId=:userId");
query.setParameter("userId", userId);
`
It dosn't work because I don't use field userId for connection. I use object User for connection.
Maybe there is another way to do this ...

Reply:
`FROM Order WHERE user.id=:userId`

Have you tried this?

Reply:
I tried
` FROM Order WHERE userId=:userId`
and it didn't work. But `FROM Order WHERE user.id=:userId` work. Thank you :-) I'll change it in the next commit.